### PR TITLE
MB-5427 Remove unused const/type/func per `unused` linter

### PR DESCRIPTION
**Pull request by felipe-lee**
## Description

Removed unused const/type/func as flagged by the `unused` linter. For some reason it doesn't pick up _a lot_ of unused things, but the scope of the current jira issue is to tackle only those flagged by the static analysis tool that was used before. We likely need another story to address the many unused things that Goland flags. 

## Setup

```sh
make server_test
```

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5427)

### CODE REVIEW

File: cmd/prime-api-dev/scripts/payment_requests.go [hunk 1]
```diff
@@ -1174,50 +1174,6 @@ func (pr *paymentRequestsData) selectMTO() error {
...
-			defer cacStore.Close()
-		}
-
-	*/
-
-	return nil
-}
-
 // updateMTOShipment creates a gateway and sends the request to the endpoint
 func (pr *paymentRequestsData) updateMTOShipment2(filename string) error {
```
Comment (sandy-wright) [inline cmd/prime-api-dev/scripts/payment_requests.go:1178]: > `updateMTOShipment2`

love to see it 😹

File: pkg/auth/authentication/auth_test.go [hunk 2]
```diff
@@ -39,35 +39,6 @@ const (
 	SddcTestHost string = "sddc.example.com"
 	// AdminTestHost
 	AdminTestHost string = "admin.example.com"
-	// FakeRSAKey generated with `bin/generate-devlocal-cert.sh -o Test -u Application -n test.mil -f test`
```
Comment (sandy-wright) [inline pkg/auth/authentication/auth_test.go:42]: Is this line documented anywhere else? I didn't know we had a script to generate fake keys, and I hope we're not deleting the only source of that info (not that it was useful here, to be fair)

**Review by sandy-wright — COMMENTED** [review]
LGTM from Roci's perspective, but I don't feel super comfortable approving it without at @jacquelineIO or someone else from A Team double-checking the invoicing deletions, and maybe someone from @transcom/truss-infrasec checking that the `ecs-deploy` command is good to go? In general, I don't think our code in the `cmd/` directory is well-tested

File: cmd/prime-api-dev/scripts/payment_requests.go (hunk 1 above)
Comment (felipe-lee) [reply cmd/prime-api-dev/scripts/payment_requests.go:1178]: Yeah haha. My guess is this one replaced the first but never got renamed.

File: pkg/auth/authentication/auth_test.go (hunk 2 above)
Comment (felipe-lee) [reply pkg/auth/authentication/auth_test.go:42]: It isn't, but also the script it references doesn't seem to exist anymore? Unless I'm missing it o.o

Comment (felipe-lee) [reply pkg/auth/authentication/auth_test.go:42]: oh i found it. got moved to the `scripts/` directory

Comment (felipe-lee) [reply pkg/auth/authentication/auth_test.go:42]: Ah, and i found the docs: https://github.com/transcom/mymove/blob/40191aca6c7ffee186a59c9918590e41b5c06bd1/config/tls/README.md#devlocal-certificate-authority

Comment (sandy-wright) [reply pkg/auth/authentication/auth_test.go:42]: NICE, thanks for looking into it. Looks like we're good to go to delete this line 👍

File: cmd/ecs-deploy/task_def.go [hunk 3]
```diff
@@ -85,14 +85,6 @@ func (e *errInvalidService) Error() string {
...
 }
 
-type errinvalidRepositoryName struct {
-	RepositoryName string
-}
-
-func (e *errinvalidRepositoryName) Error() string {
-	return fmt.Sprintf("invalid AWS ECR respository name %q", e.RepositoryName)
-}
-
```
Comment (kilbergr) [inline cmd/ecs-deploy/task_def.go:95]: I agree this isn't used anywhere, so think it is fine to remove!

**Review by kilbergr — APPROVED** [review]
Infram ok!

**Review by kilbergr — COMMENTED** [review]
Sorry--meant to comment that--from an infra perspective ok, but leaving for Jacquie and others to 👀

File: cmd/prime-api-dev/scripts/payment_requests.go (hunk 1 above)
Comment (jacquelineIO) [reply cmd/prime-api-dev/scripts/payment_requests.go:1178]: @sandy-wright it's not real code unless there are some `*2` `*3` functions somewhere lol

**Review by jacquelineIO — APPROVED** [review]

**Pull request merged**